src/Services/DSCFileActivationHandler.cs

```diff
@@ -0,0 +1,103 @@
+
+namespace DevHome.Services;
+
+/// <summary>
+/// Class that handles the activation of the application when a DSC file (*.winget) is opened
+/// </summary>
+public class DSCFileActivationHandler : ActivationHandler<FileActivatedEventArgs>
+{
+    private const string WinGetFileExtension = ".winget";
+    private readonly ISetupFlowStringResource _setupFlowStringResource;
+    private readonly INavigationService _navigationService;
+    private readonly SetupFlowViewModel _setupFlowViewModel;
+    private readonly SetupFlowOrchestrator _setupFlowOrchestrator;
+    private readonly WindowEx _mainWindow;
+
+    public DSCFileActivationHandler(
+        ISetupFlowStringResource setupFlowStringResource,
+        INavigationService navigationService,
+        SetupFlowOrchestrator setupFlowOrchestrator,
+        SetupFlowViewModel setupFlowViewModel,
+        WindowEx mainWindow)
+    {
+        _setupFlowStringResource = setupFlowStringResource;
+        _setupFlowViewModel = setupFlowViewModel;
+        _setupFlowOrchestrator = setupFlowOrchestrator;
+        _navigationService = navigationService;
+        _mainWindow = mainWindow;
+    }
+
+    protected override bool CanHandleInternal(FileActivatedEventArgs args)
+    {
+        return args.Files.Count > 0 && args.Files[0] is StorageFile file && file.FileType == WinGetFileExtension;
+    }
+
+    protected async override Task HandleInternalAsync(FileActivatedEventArgs args)
+    {
+        Debug.Assert(_navigationService.Frame != null, "Main window content is expected to be set before activation handlers are executed");
+        var file = (StorageFile)args.Files[0];
+        async void DSCActivationFlowHandlerAsync(object sender, RoutedEventArgs e)
+        {
+            // Only execute once
+            _navigationService.Frame!.Loaded -= DSCActivationFlowHandlerAsync;
+            await DSCActivationFlowAsync(file);
+        }
+
+        // If the application was activated from a file, the XamlRoot here is null
+        if (_navigationService.Frame.XamlRoot == null)
+        {
+            // Wait until the frame is loaded before starting the flow
+            Log.Logger?.ReportInfo("DSC flow activated from a file but the application is not yet ready. Activation will start once the page is loaded.");
+            _navigationService.Frame!.Loaded += DSCActivationFlowHandlerAsync;
+        }
+        else
+        {
+            // If the application was already running, start the flow immediately
+            await DSCActivationFlowAsync(file);
+        }
+    }
+
+    /// <summary>
+    /// Navigates to the setup flow and starts the DSC activation flow
+    /// </summary>
+    /// <param name="file">The DSC file to activate</param>
+    private async Task DSCActivationFlowAsync(StorageFile file)
+    {
+        try
+        {
+            // Don't interrupt the user if the machine configuration is in progress
+            if (_setupFlowOrchestrator.FlowPages.Count > 1)

```

Comment:
Implement `IsActive` property.
```suggestion
            if (_setupFlowOrchestrator.IsActive)
```